runtime/onert/core/src/exec/ExecutorBase.cc

```diff
@@ -63,7 +63,7 @@ void ExecutorBase::execute(const std::vector<backend::IPortableTensor *> &inputs
   for (uint32_t n = 0; n < inputs.size(); ++n)
   {
     const auto input = inputs[n];
-    assert(input->buffer() != nullptr);
+    assert(input->buffer() != nullptr || input->get_info().total_size() == 0);
```

Comment:
```suggestion
    assert(input->buffer() != nullptr && input->get_info().total_size() != 0);
```

may be..?

Reply:
No. It should pass when tensor is general or optional
- General: `input->buffer() != nullptr`
- Optional: `input->buffer() == nullptr && input->get_info().total_size() == 0`

-> `(input->buffer() != nullptr  || (input->buffer() == nullptr && input->get_info().total_size() == 0))`
-> `(input->buffer() != nullptr  | input->get_info().total_size() == 0)`